src/pages/Staff/components/ProductsTable/ProductsTableRow.tsx

```diff
@@ -16,12 +17,13 @@ export default function ProductsTableRow({ product, formatting = '' }: Props): R
   const { removeProduct } = useProducts();
   const productForm = useProductFormContext();
 
-  const handleDelete = (): void => {
-    removeProduct.mutate(product.id, {
-      onSuccess: () => {
-        toast.success(`Product "${product.name}" was succesfully removed.`);
-      },
-    });
+  const handleDelete = async (): Promise<void> => {
+    try {
+      await removeProduct.mutateAsync(product.id);
+      toast.success(`The Product, ${product.name}, was successfully retired.`);
```

Comment:
im not too aware of how `mutateAsync` works, will that always throw an exception if the request fails?